### PR TITLE
Signup

### PR DESCRIPTION
**Pull request by jaynopponep**
signup feature correctly implemented
any errors should be displayed when occurred, but should be running smoothly

### CODE REVIEW

**Review by Abdul-Andha — COMMENTED** [review]
Looks good just needs some small changes

File: client/src/app/signup/actions.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
-  } catch (error) {
-    throw error;
+  const response = await fetch('http://localhost:8080/api/v1/users/register', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(formData)
+  });
+  if (!response.ok) {
```
[the remark this answers is not in the document]
Comment (jaynopponep) [reply client/src/app/signup/actions.js]: done, let me know if i did it incorrectly

File: client/src/app/signup/actions.js [hunk 2]
```diff
@@ -2,16 +2,22 @@
...
+      },
+      body: JSON.stringify(formData)
+    });
+    const data = await response.json();
+    if (!response.ok) {
+      console.error('Registration failed: ', data.error || data); // Use 'data.error' based on your backend error key
+      throw new Error(data.error || 'Registration failed');  // Use the error from the backend if available
     }
-    return testUser
+    console.log('Successful Registration:', data);
```
Comment (Abdul-Andha) [inline client/src/app/signup/actions.js:17]: we dont want to keep the console.logs in the final product

File: server/app/api/users.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    response = test_db_connection()
+    return jsonify(response)
+
+
+@users_api_v1.route('/register', methods=['POST'])
+def api_register():
+    if not request.is_json:
+        return jsonify({'error': 'Not JSON request'}), 400
+    request_data = request.get_json()
+    print("received: ", request_data)
```
Comment (Abdul-Andha) [inline server/app/api/users.py:31]: same with prints

**Review by Abdul-Andha — COMMENTED** [review]
Everything looks good just remove any console.logs and prints statements

**Comment by Abdul-Andha** [comment]
Everything looks good. Im merging this to main

**Pull request merged**